### PR TITLE
Add `esbuild-vue` package types

### PR DESCRIPTION
**Pull request by iam-medvedev**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/esbuild-vue/tsconfig.json
Consider running `npx dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8fe71951a3686bac00fea62fd7c9ea0413f9c3ae
</p>

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/esbuild-vue/tsconfig.json
Consider running `npx dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7a6349baa3d2683bb9a2b4c9eed0082452bf718f
</p>

**Review by esafev — APPROVED** [review]

File: types/esbuild-vue/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        "esModuleInterop": true,
+        "strictNullChecks": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true
+    },
```
Comment (jakebailey) [inline types/esbuild-vue/tsconfig.json:19]: Why have you removed `esModuleInterop`? See the bot's comment.

Comment (iam-medvedev) [reply types/esbuild-vue/tsconfig.json:19]: Hmm, but it's [here](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67017/commits/7a6349baa3d2683bb9a2b4c9eed0082452bf718f#diff-99064c37cfd03a0fde18694cb8b3297ca22876636bf6c6921d26c658e211ad77R10).

**Review by esafev — APPROVED** [review]

File: types/esbuild-vue/tsconfig.json (hunk 1 above)
Comment (jakebailey) [reply types/esbuild-vue/tsconfig.json:19]: Hm, the bot's still unhappy with it, for some reason. Maybe it needs to be gone entirely?

Comment (iam-medvedev) [reply types/esbuild-vue/tsconfig.json:18]: That's right. It's okay now.

**Review by jakebailey — APPROVED** [review]

**Comment by iam-medvedev** [comment]
Ready to merge

**Pull request merged**